packages/grafana-ui/src/components/Card/Card.tsx

```diff
@@ -49,95 +47,82 @@ const CardInner: FC<CardInnerProps> = ({ children, href }) => {
-  /** Image or icon to be displayed on the let side of the card */
-  image?: ReactNode;
-  /** Main card actions **/
-  actions?: ReactElement[];
-  /** Right-side actions */
-  secondaryActions?: ReactElement[];
   /** Link to redirect to on card click. If provided, the Card inner content will be rendered inside `a` */
   href?: string;
   /** On click handler for the Card */
   onClick?: () => void;
 }
 
+export interface CardInterface extends FC<Props> {
+  Tags: typeof Tags;
+  Figure: typeof Figure;
+  Meta: typeof Meta;
+  Actions: typeof Actions;
+  SecondaryActions: typeof SecondaryActions;
+}
+
 /**
  * Generic card component
  *
  * @alpha
  */
-export const Card: FC<Props> = ({
+export const Card: CardInterface = ({
   heading,
   description,
-  metadata,
-  tags = [],
-  onTagClick,
   disabled,
-  image,
-  actions = [],
   tooltip,
-  secondaryActions = [],
   href,
   onClick,
   className,
+  children,
   ...htmlProps
 }) => {
-  const hasActions = Boolean(actions.length || secondaryActions.length);
-  const disableHover = disabled || actions.length > 1 || !onClick;
-  const disableEvents = disabled && !actions.length;
   const theme = useTheme();
-  const styles = getCardStyles(theme, disableEvents, disableHover);
-  // Join meta data elements by '|'
-  const meta = useMemo(
-    () =>
-      Array.isArray(metadata)
-        ? (metadata as ReactNode[]).filter(Boolean).reduce((prev, curr, i) => [
-            prev,
-            <span key={`separator_${i}`} className={styles.separator}>
-              |
-            </span>,
-            curr,
-          ])
-        : metadata,
-    [metadata, styles.separator]
+  const styles = getCardStyles(theme);
+  const [tags, figure, meta, actions, secondaryActions] = ['Tags', 'Figure', 'Meta', 'Actions', 'SecondaryActions'].map(
+    item => {
+      //@ts-ignore
+      const found = React.Children.toArray(children).find((child: JSX.Element) => {
+        return child?.type?.displayName === item;
+      }) as JSX.Element | undefined;
+
+      if (found) {
+        return React.cloneElement(found, { ...found.props, styles, disabled });
+      }
+      return found;
+    }
   );
+
+  const hasActions = Boolean(actions || secondaryActions);
+  const disableHover = disabled || !onClick;
+  const disableEvents = disabled && !actions;
+
+  const containerStyles = getContainerStyles(theme, disableEvents, disableHover);
   const onCardClick = useCallback(() => (disableHover ? () => {} : onClick), [disableHover, onClick]);
 
   return (
     <CardContainer
       tooltip={tooltip}
       tabIndex={disableHover ? undefined : 0}
-      className={cx(styles.container, className)}
+      className={cx(containerStyles, className)}
       onClick={onCardClick}
       {...htmlProps}
     >
       <CardInner href={href}>
-        {image && <div className={styles.media}>{image}</div>}
+        {figure}
         <div className={styles.inner}>
           <div className={styles.heading}>{heading}</div>
```

Comment:
Might be a good idea to add the "heading" role here
```suggestion
          <div className={styles.heading} role="heading">
            {heading}
          </div>
```

Reply:
Great catch!